db2-plugin/src/test/java/co/cask/db2/Db2SourceTestRun.java

```diff
@@ -0,0 +1,326 @@
+    etlConfig = ETLBatchConfig.builder()
+      .addStage(database)
+      .addStage(table)
+      .addConnection(database.getName(), table.getName())
+      .build();
+    appRequest = new AppRequest<>(DATAPIPELINE_ARTIFACT, etlConfig);
+    deployApplication(appId, appRequest);
+  }
+
+  @Test
+  public void testNonExistentDBTable() throws Exception {
+    // source
+    String importQuery = "SELECT ID, NAME FROM dummy WHERE ID < 3 AND $CONDITIONS";
+    String boundingQuery = "SELECT MIN(ID),MAX(ID) FROM dummy";
+    String splitBy = "ID";
+    ETLPlugin sinkConfig = MockSink.getPlugin("table");
+    ETLPlugin sourceBadNameConfig = new ETLPlugin(
+      UI_NAME,
+      BatchSource.PLUGIN_TYPE,
+      ImmutableMap.<String, String>builder()
+        .putAll(BASE_PROPS)
+        .put(AbstractDBSource.DBSourceConfig.IMPORT_QUERY, importQuery)
+        .put(AbstractDBSource.DBSourceConfig.BOUNDING_QUERY, boundingQuery)
+        .put(AbstractDBSource.DBSourceConfig.SPLIT_BY, splitBy)
+        .put(Constants.Reference.REFERENCE_NAME, "DBNonExistentTest")
+        .build(),
+      null);
+    ETLStage sink = new ETLStage("sink", sinkConfig);
+    ETLStage sourceBadName = new ETLStage("sourceBadName", sourceBadNameConfig);
+
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder()
+      .addStage(sourceBadName)
+      .addStage(sink)
+      .addConnection(sourceBadName.getName(), sink.getName())
+      .build();
+    ApplicationId appId = NamespaceId.DEFAULT.app("dbSourceNonExistingTest");
+    assertRuntimeFailure(appId, etlConfig, DATAPIPELINE_ARTIFACT,
+                         "ETL Application with DB Source should have failed because of a " +
+      "non-existent source table.", 1);
+
+    // Bad connection
+    ETLPlugin sourceBadConnConfig = new ETLPlugin(
+      UI_NAME,
+      BatchSource.PLUGIN_TYPE,
+      ImmutableMap.<String, String>builder()
+        .put("host", BASE_PROPS.get("host"))
+        .put("port", BASE_PROPS.get("port"))
+        .put("database", "dumDB")
+        .put("user", BASE_PROPS.get("user"))
+        .put("password", BASE_PROPS.get("password"))
+        .put("jdbcPluginName", JDBC_DRIVER_NAME)
+        .put(AbstractDBSource.DBSourceConfig.IMPORT_QUERY, importQuery)
+        .put(AbstractDBSource.DBSourceConfig.BOUNDING_QUERY, boundingQuery)
+        .put(AbstractDBSource.DBSourceConfig.SPLIT_BY, splitBy)
+        .put(Constants.Reference.REFERENCE_NAME, "DB2Test")
+        .build(),
+      null);
+    ETLStage sourceBadConn = new ETLStage("sourceBadConn", sourceBadConnConfig);
+    etlConfig = ETLBatchConfig.builder()
+      .addStage(sourceBadConn)
+      .addStage(sink)
+      .addConnection(sourceBadConn.getName(), sink.getName())
+      .build();
+    assertRuntimeFailure(appId, etlConfig, DATAPIPELINE_ARTIFACT,
+                         "ETL Application with DB Source should have failed because of a " +
```

Comment:
Is this an error message hard coded somewhere? If so can we extract to a protected constant and mark as @VisibleForTesting

Reply:
No, it's only used in tests.